sos/report/plugins/gcp.py

```diff
@@ -0,0 +1,155 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+import json
+from http.client import HTTPResponse
+from typing import Any
+from urllib import request
+from urllib.error import URLError
+
+from sos.report.plugins import Plugin, IndependentPlugin, PluginOpt
+from sos.report.plugins import SoSPredicate
+
+
+class GCP(Plugin, IndependentPlugin):
+
+    short_desc = 'Google Cloud Platform'
+
+    plugin_name = 'gcp'
+    profiles = ('virt',)
+
+    option_list = [
+        PluginOpt('keep-pii', default=False,
+                  desc="Stop the plugin from removing PIIs like project name "
+                       "or organization ID from the metadata retrieved from "
+                       "Metadata server.")
+    ]
+
+    METADATA_ROOT = "http://metadata.google.internal/computeMetadata/v1/"
+    METADATA_QUERY = "http://metadata.google.internal/computeMetadata/v1/" \
+                     "?recursive=true"
+    REDACTED = "[--REDACTED--]"
+
+    # A line we will be looking for in the dmesg output. If it's there,
+    # that means we're running on a Google Cloud Compute instance.
+    GOOGLE_DMI = "DMI: Google Google Compute Engine/Google " \
+                 "Compute Engine, BIOS Google"
+
+    def check_enabled(self):
+        """
+        Checks if this plugin should be executed at all. In this case, it
+        will check the `dmesg` command output to see if the system is
+        running on a Google Cloud Compute instance.
+        """
+        dmesg = self.exec_cmd("dmesg")
+        if dmesg['status'] != 0:
+            return False
+        return self.GOOGLE_DMI in dmesg['output']
+
+    def _has_gcloud_snap(self) -> bool:
+        return self.exec_cmd("snap list google-cloud-sdk")["status"] == 0
+
+    def setup(self):
+        """
+        Collect the following info:
+         * Metadata from the Metadata server
+         * `gcloud auth list` output
+         * Any google services output from journal
+        """
+
+        # Capture gcloud auth list
+        if self._has_gcloud_snap():
+            self.add_cmd_output("gcloud auth list", tags=['gcp'])
+        else:
+            predicate = SoSPredicate(self, packages=['google-cloud-sdk'])
+            self.add_cmd_output("gcloud auth list", pred=predicate,
+                                tags=['gcp'])
```

Comment:
`add_cmd_output()` automatically gracefully handles when a called command does not exist. So this can be a "plain" `self.add_cmd_output("gcloud auth list", tags=['gcp'])` and it will function properly regardless of where (or if) `gcloud` is installed from.

Reply:
OK, I'll just switch to the "plain" version you described. Makes life easier, thanks :)